File: 1103/1019.cpp
#include<cstdio>
#include<cstring>
#include<vector>

using namespace std;

vector<int> f(int x)
{
    vector<int> result(10);

    if (x <= 9) {
        for (int i = 1; i <= x; i++)
            result[i] = 1;
        return result;
    }

    while (x % 10 != 9) {
        int t = x;
        while (t) {
            result[t % 10]++;
            t /= 10;
        }
        x--;
    }

    vector<int> div10 = f(x / 10);
    for (int i = 0; i < 10; i++)
        result[i] += div10[i] * 10 + (x / 10 + (i > 0));

    return result;
}

int main()
{
    int n;
    scanf("%d", &n);

    vector<int> result = f(n);
    for (int i = 0; i < 10; i++)
        printf("%d ", result[i]);
    printf("\n");

    return 0;
}

File: 1109/1019.java
import java.util.*;
import java.lang.*;

public class Main {
    Scanner sc = new Scanner(System.in);
    int n;

    int[] f(int n)
    {
        int[] result = new int[10];

        while (n > 0 && n % 10 != 9) {
            int t = n;
            while (t > 0) {
                result[t % 10]++;
                t /= 10;
            }
            n--;
        }

        if (n == 0) return result;

        int[] div10 = f(n / 10);
        for (int i = 0; i < 10; i++)
            result[i] += div10[i] * 10 + (n / 10) + 1;
        result[0]--;

        return result;
    }

    public void run()
    {
        n = sc.nextInt();
        int[] ans = f(n);

        for (int i = 0; i < 10; i++)
            System.out.print(ans[i] + " ");
        System.out.println();
    }

    public static void main(String[] args)
    {
        new Main().run();
    }
}
